### PR TITLE
Better document unsafe inputs in the QueryBuilder.

### PR DESCRIPTION
**Pull request by markstory**
Better document which methods are not safe for untrusted data.

Refs #11148

### CODE REVIEW

File: src/Database/Query.php [hunk 1]
```diff
@@ -1076,6 +1082,9 @@ public function order($fields, $overwrite = false)
      * This method allows you to set complex expressions
      * as order conditions unlike order()
      *
+     * Order fields are not suitable for use with user supplied data as they are
```
Comment (saeideng) [inline src/Database/Query.php]: plain text or text in note box(warning,..) for these notes?

Comment (lorenzo) [reply src/Database/Query.php]: Is it possible to make a box here?

Comment (markstory) [reply src/Database/Query.php]: I don't think so unless we start embedding HTML in docblocks.

Comment (saeideng) [reply src/Database/Query.php]: oh, I forgot  this repo is for `core` not `docs` ,i'm sorry :)

**Pull request merged**